### PR TITLE
feat: source configuration for temporal types

### PR DESCRIPTION
**Pull request by Emrehzl94**
This pr implements a mechanism to configure temporal data schema types for source connector and extends test module for Kafka consumer.

### CODE REVIEW

**Review by venikkin — APPROVED** [review]
Looks good. Have one comment.

File: testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/kafka/ConsumerResolver.kt [hunk 1]
```diff
@@ -41,36 +42,36 @@ internal class ConsumerResolver(
...
-    val valueConverter = keyValueConverterResolver.resolveValueConverter(extensionContext)
-    if (keyConverter.supportsSchemaRegistry || valueConverter.supportsSchemaRegistry) {
-      properties.setProperty(
-          KafkaAvroDeserializerConfig.SCHEMA_REGISTRY_URL_CONFIG,
-          schemaControlRegistryExternalUriProvider(),
-      )
-    }
+
+    properties.setProperty(
+        AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
```
[the remark this answers is not in the document]
Comment (Emrehzl94) [reply testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/kafka/ConsumerResolver.kt:62]: yes, you're right. Thanks!

File: testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/assertions/TopicVerifier.kt [hunk 2]
```diff
@@ -91,11 +109,74 @@ class TopicVerifier<K, V>(
...
+        .fields()
+        .filter { struct.get(it) != null }
+        .forEach { field ->
+          when (field.schema().type()) {
+            Schema.Type.STRUCT -> map[field.name()] = structToMap(struct.getStruct(field.name()))
+            Schema.Type.ARRAY -> map[field.name()] = convertList(struct.getArray<Any>(field.name()))
+            else -> map[field.name()] = struct.get(field)
+          }
+        }
+    return map
```
Comment (ali-ince) [inline testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/assertions/TopicVerifier.kt:143]: Could we make use of DynamicTypes in here?

**Review by ali-ince — APPROVED** [review]
Looks good to me.

File: testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/assertions/TopicVerifier.kt (hunk 2 above)
Comment (Emrehzl94) [reply testing/src/main/kotlin/org/neo4j/connectors/kafka/testing/assertions/TopicVerifier.kt:143]: yes, it's better with DynamicTypes. Thanks!

**Pull request merged**